core/src/main/java/jenkins/management/ShutdownLink.java

```diff
@@ -27,35 +27,54 @@
 import hudson.Extension;
 import hudson.model.ManagementLink;
 import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
 import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.verb.POST;
 
-/**
- * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
- */
-@Extension(ordinal = Integer.MIN_VALUE) @Symbol("shutDown")
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Extension(ordinal = Integer.MIN_VALUE)
+@Symbol("prepareShutdown")
 public class ShutdownLink extends ManagementLink {
 
+    private static final Logger LOGGER = Logger.getLogger(ShutdownLink.class.getName());
+
     @Override
     public String getIconFileName() {
         return "system-log-out.png";
     }
 
     public String getDisplayName() {
-        return Jenkins.get().isQuietingDown() ? Messages.ShutdownLink_DisplayName_cancel() : Messages.ShutdownLink_DisplayName_prepare();
+        return Messages.ShutdownLink_DisplayName_prepare();
     }
 
     @Override
     public String getDescription() {
-        return Jenkins.get().isQuietingDown() ? "" : Messages.ShutdownLink_Description();
+        return Messages.ShutdownLink_Description();
     }
 
     @Override
     public String getUrlName() {
-        return Jenkins.get().isQuietingDown() ? "cancelQuietDown" : "quietDown";
+        return "prepareShutdown";
     }
 
-    @Override
-    public boolean getRequiresPOST() {
-        return true;
+    @POST
+    public synchronized void doPrepare(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, InterruptedException {
```

Comment:
Lacks permission check. While `#doQuietDown` does, this logs messages before permission is checked.

Reply:
I think it's quite reasonable logging before permission check. If user don't have permission - exception will be logged. But in any situation(user have permission or not) we will log that action was `requested`. It can help in problems investigations. But if you insist I can change it.

Reply:
What is logged if the user doesn't have permissions? Any kind of information that might be private?

Reply:
Permission check added